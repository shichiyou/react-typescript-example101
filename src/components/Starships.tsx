import React from 'react';
import useStarshipsService from '../services/useStarshipsService';

const Starships: React.FC<{}> = () => {
  const service = useStarshipsService();

  return (
    <div>
      {service.status === 'loading' && <div>Loading...</div>}
      {service.status === 'loaded' &&
        service.payload.results.map(starship => (
          <div key={starship.url}>{starship.name}</div>
        ))}
      {service.status === 'error' && (
        <div>Error, the backend moved to the dark side.</div>
      )}
    </div>
  );
};

export default Starships;
