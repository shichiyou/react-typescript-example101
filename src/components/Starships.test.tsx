import React from 'react';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { shallow, mount, render } from 'enzyme';
import { ShallowWrapper, ReactWrapper } from 'enzyme';
import Starships from './Starships';
import fetchMock from 'fetch-mock';
import responseData from '../services/Starships.json';
import toJson from 'enzyme-to-json';

describe('<Starships />', () => {
  let container: HTMLDivElement | Element | null;

  beforeAll(() => {
    fetchMock.get('https://swapi.co/api/starships', JSON.stringify(responseData));
  });
  afterAll(() => {
    fetchMock.restore();
  });

  test("it shows a list of starships", async () => {
    let wrapper: ReactWrapper;

    // Test first render and effect
    act(() => {
      wrapper = mount(<Starships />);
      expect(wrapper!.text()).toBe('Loading...');
      wrapper!.update();
      expect(toJson(wrapper!)).toMatchSnapshot('step 2');
      wrapper!.update();
      expect(toJson(wrapper!)).toMatchSnapshot('step 3');
      wrapper!.update();
      expect(toJson(wrapper!)).toMatchSnapshot('step 4');
      wrapper!.update();
    });



    wrapper!.unmount();
});  
});
