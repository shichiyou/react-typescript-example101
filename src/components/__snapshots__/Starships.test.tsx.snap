// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Starships /> can render and update a Starships 1`] = `ReactWrapper {}`;

exports[`<Starships /> it shows a list of starships 1`] = `<div />`;

exports[`<Starships /> it shows a list of starships: step 1 1`] = `<div />`;

exports[`<Starships /> it shows a list of starships: step 2 1`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;

exports[`<Starships /> it shows a list of starships: step 2 2`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;

exports[`<Starships /> it shows a list of starships: step 3 1`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;

exports[`<Starships /> it shows a list of starships: step 4 1`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;

exports[`<Starships /> it shows a list of starships: step XS 1`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;

exports[`<Starships /> renders correctly enzyme 1`] = `
<Starships>
  <div>
    <div>
      Loading...
    </div>
  </div>
</Starships>
`;
